File: python/card_batch.py
"""Submit a card-pricing batch and poll until it reaches a terminal state."""

from __future__ import annotations

import os
import time
import uuid

import requests

API_BASE = "https://api.compsniper.com"
TERMINAL = {"done", "partial", "cancelled", "failed"}


def required_api_key() -> str:
    key = os.environ.get("COMPSNIPER_API_KEY", "").strip()
    if not key:
        raise SystemExit("Set COMPSNIPER_API_KEY before running this example.")
    return key


headers = {
    "Authorization": f"Bearer {required_api_key()}",
    "Content-Type": "application/json",
    "Idempotency-Key": f"example-{uuid.uuid4()}",
}
payload = {
    "cards": [
        {
            "reference": "inventory-001",
            "year": 2023,
            "set": "Topps Chrome",
            "player": "Victor Wembanyama",
            "cardNumber": "1",
            "parallel": "Refractor",
            "grader": "PSA",
            "grade": 10,
        },
        {
            "reference": "inventory-002",
            "keyword": "1999 pokemon base set charizard 4/102 psa 9",
        },
    ],
    "options": {"count": 60, "outputMode": "summary", "relevance": True},
}

submitted = requests.post(
    f"{API_BASE}/v1/cards/batch",
    headers=headers,
    json=payload,
    timeout=30,
)
submitted.raise_for_status()
job = submitted.json()
job_id = job["jobId"]
print("Submitted job:", job_id)

while True:
    response = requests.get(
        f"{API_BASE}/v1/cards/batch/{job_id}",
        headers={"Authorization": headers["Authorization"]},
        timeout=30,
    )
    response.raise_for_status()
    job = response.json()
    print(f"{job['status']}: {job['processedCards']}/{job['totalCards']} processed")
    if job["status"] in TERMINAL:
        break
    time.sleep(5)

for result in job.get("results", []):
    print(result.get("reference"), result.get("status"), result.get("summary"))

File: node/card-batch.mjs
import { randomUUID } from "node:crypto";

const API_BASE = "https://api.compsniper.com";
const TERMINAL = new Set(["done", "partial", "cancelled", "failed"]);
const apiKey = process.env.COMPSNIPER_API_KEY?.trim();
if (!apiKey) throw new Error("Set COMPSNIPER_API_KEY before running this example.");

const authorization = `Bearer ${apiKey}`;
const submitted = await fetch(`${API_BASE}/v1/cards/batch`, {
  method: "POST",
  headers: {
    Authorization: authorization,
    "Content-Type": "application/json",
    "Idempotency-Key": `example-${randomUUID()}`,
  },
  body: JSON.stringify({
    cards: [
      {
        reference: "inventory-001",
        year: 2023,
        set: "Topps Chrome",
        player: "Victor Wembanyama",
        cardNumber: "1",
        parallel: "Refractor",
        grader: "PSA",
        grade: 10,
      },
      { reference: "inventory-002", keyword: "1999 pokemon base set charizard 4/102 psa 9" },
    ],
    options: { count: 60, outputMode: "summary", relevance: true },
  }),
  signal: AbortSignal.timeout(30_000),
});
let job = await submitted.json();
if (!submitted.ok) throw new Error(job.error ?? `CompSniper returned HTTP ${submitted.status}`);
console.log("Submitted job:", job.jobId);

while (!TERMINAL.has(job.status)) {
  await new Promise((resolve) => setTimeout(resolve, 5000));
  const response = await fetch(`${API_BASE}/v1/cards/batch/${job.jobId}`, {
    headers: { Authorization: authorization },
    signal: AbortSignal.timeout(30_000),
  });
  job = await response.json();
  if (!response.ok) throw new Error(job.error ?? `CompSniper returned HTTP ${response.status}`);
  console.log(`${job.status}: ${job.processedCards}/${job.totalCards} processed`);
}

for (const result of job.results ?? []) {
  console.log(result.reference, result.status, result.summary);
}
